app/importers/unpaywall_publication_importer.rb

```diff
@@ -0,0 +1,91 @@
+# frozen_string_literal: true
+
+class UnpaywallPublicationImporter
+  def import_all
+    unless Rails.env.test?
+      pbar = ProgressBar.create(title: 'Importing publication data from Unpaywall',
+                                total: all_pubs.count)
+    end
+
+    all_pubs.find_each do |p|
+      query_unpaywall_for(p)
+      pbar.increment unless Rails.env.test?
+    end
+    pbar.finish unless Rails.env.test?
+  end
+
+  def import_new
+    unless Rails.env.test?
+      pbar = ProgressBar.create(title: 'Importing publication data from Unpaywall',
+                                total: new_pubs.count)
+    end
+
+    new_pubs.find_each do |p|
+      query_unpaywall_for(p)
+      pbar.increment unless Rails.env.test?
+    end
+    pbar.finish unless Rails.env.test?
+  end
+
+  private
+
+    def all_pubs
+      Publication.where.not(doi: nil).where.not(doi: '')
+    end
+
+    def new_pubs
+      all_pubs.where(unpaywall_last_checked_at: nil)
+    end
+
+    def get_pub(url)
+      attempts = 0
+      HTTParty.get(url).to_s
+    rescue Net::ReadTimeout, Net::OpenTimeout
```

Comment:
This block isn't getting tested, and seems worthy enough. What do you think? I can give you a hand if you need.

Reply:
Great call to test this, because it was broken 😁 

`attempts` kept getting reset to 0 at the beginning of the function so it would keep attempting the API call in an infinite loop. We'll have to make this same change in the `open_access_button_importer` as well. I will open a ticket for fixing that + other improvements based on the feedback from this PR.

Reply:
Yeah, that's a great idea. What is the convention for declaring/importing "utility methods" in Rails? I would think we could rename this method to something like `HTTPHelper::try_get(url)` or something like that. But definitely open to other naming suggestions.